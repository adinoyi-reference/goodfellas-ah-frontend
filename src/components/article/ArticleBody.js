import React from 'react';
import parser from 'react-html-parser';
import SideButtons from './SideButtons';


const ArticleBody = (props) => {
  const { article, userId } = props;
  const { body } = article;
  return (
    <div className="article-container">
      <SideButtons article={article} userId={userId} />
      <div className="article-content">
        {parser(body)}
      </div>
    </div>
  );
};


export default ArticleBody;
