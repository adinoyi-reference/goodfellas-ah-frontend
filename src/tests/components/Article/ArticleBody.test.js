import React from 'react';
import { shallow } from 'enzyme';
import ArticleBody from '../../../components/article/ArticleBody';

const setup = () => {
  const props = {
    article: { authorId: 1, image: 'thhisimage', body: 'somerandombody' },
    userId: 1,
  };

  const enzymeWrapper = shallow(<ArticleBody {...props} />);

  return {
    props,
    enzymeWrapper,
  };
};

describe('components', () => {
  describe('ArticleBody component', () => {
    it('should render as expected', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
