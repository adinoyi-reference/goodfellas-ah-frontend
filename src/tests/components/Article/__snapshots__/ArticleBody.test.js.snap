// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ArticleBody component should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleBody
    article={
      Object {
        "authorId": 1,
        "body": "somerandombody",
        "image": "thhisimage",
      }
    }
    userId={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SideButtons
          article={
            Object {
              "authorId": 1,
              "body": "somerandombody",
              "image": "thhisimage",
            }
          }
          userId={1}
        />,
        <div
          className="article-content"
        >
          somerandombody
        </div>,
      ],
      "className": "article-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "article": Object {
            "authorId": 1,
            "body": "somerandombody",
            "image": "thhisimage",
          },
          "userId": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "somerandombody",
          ],
          "className": "article-content",
        },
        "ref": null,
        "rendered": Array [
          "somerandombody",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideButtons
            article={
              Object {
                "authorId": 1,
                "body": "somerandombody",
                "image": "thhisimage",
              }
            }
            userId={1}
          />,
          <div
            className="article-content"
          >
            somerandombody
          </div>,
        ],
        "className": "article-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "article": Object {
              "authorId": 1,
              "body": "somerandombody",
              "image": "thhisimage",
            },
            "userId": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "somerandombody",
            ],
            "className": "article-content",
          },
          "ref": null,
          "rendered": Array [
            "somerandombody",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
